### PR TITLE
Ignore def pattern for function args

### PR DESCRIPTION
**Pull request by wontem**
Part of https://github.com/roc-lang/roc/issues/6765

### CODE REVIEW

**Review by lukewilliamboswell — DISMISSED** [review]
Nice work 😀

File: crates/compiler/can/tests/test_suffixed.rs [hunk 1]
```diff
@@ -804,6 +804,38 @@ mod suffixed_tests {
...
+                c
+            "#,
+            r##"Defs { tags: [Index(2147483648)], regions: [@0-49], space_before: [Slice(start = 0, length = 0)], space_after: [Slice(start = 0, length = 0)], spaces: [], type_defs: [], value_defs: [Body(@0-4 Identifier { ident: "main" }, @27-31 Apply(@27-31 Var { module_name: "Task", ident: "await" }, [@29-30 Var { module_name: "", ident: "a" }, @27-31 Closure([@29-30 Identifier { ident: "#!a0" }], @27-31 Defs(Defs { tags: [Index(2147483650)], regions: [@27-31], space_before: [Slice(start = 0, length = 0)], space_after: [Slice(start = 0, length = 0)], spaces: [], type_defs: [], value_defs: [Body(@23-24 Identifier { ident: "c" }, @27-31 Apply(@27-28 Var { module_name: "", ident: "b" }, [@29-30 TaskAwaitBang(Var { module_name: "", ident: "a" })], Space)), Body(@23-24 Identifier { ident: "c" }, @27-31 Apply(@27-28 Var { module_name: "", ident: "b" }, [@29-30 Var { module_name: "", ident: "#!a0" }], Space)), Body(@23-24 Identifier { ident: "c" }, @27-31 Apply(@27-28 Var { module_name: "", ident: "b" }, [@29-30 Var { module_name: "", ident: "#!a0" }], Space))] }, @48-49 Var { module_name: "", ident: "c" }))], BangSuffix))] }"##,
```
Comment (wontem) [inline crates/compiler/can/tests/test_suffixed.rs:820]: Hmm, too many definitions inside of the closure, need to drop the first one I guess. Will take a closer look tomorrow.

Comment (wontem) [reply crates/compiler/can/tests/test_suffixed.rs:820]: The same problem in some other tests. Must be smth in `replace_with_value_def`

Comment (lukewilliamboswell) [reply crates/compiler/can/tests/test_suffixed.rs:820]: It's an optimisation - deliberately left there if it is what I think it is. I got stumped on that for weeks.

Comment (wontem) [reply crates/compiler/can/tests/test_suffixed.rs:820]: yeah, I just figured we only push but never delete from the collection. and `tags` are the source of truth. need to add a comment about it. I'll add as a follow-up

**Review by lukewilliamboswell — APPROVED** [review]
Looks good thank you 😍

**Pull request merged**